packages/tween/src/index.ts

```diff
@@ -21,21 +21,32 @@ export default function createTween<T extends number>(
     return target;
   }
 
-  const [start, setStart] = createSignal(performance.now());
   const [current, setCurrent] = createSignal<T>(target());
-  createEffect(on(target, () => setStart(performance.now()), { defer: true }));
-  createEffect(
-    on([start, current], () => {
-      const cancelId = requestAnimationFrame(t => {
-        const elapsed = t - (start() || 0) + 1;
-        // @ts-ignore
-        setCurrent(c =>
-          elapsed < duration ? (target() - c) * ease((elapsed / duration) as T) + c : target(),
-        );
-      });
-      onCleanup(() => cancelAnimationFrame(cancelId));
-    }),
-  );
+  let startValue = target();
+  let start = 0;
+  let cancelId: number;
+
+  function tick(t: number) {
+    const elapsed = t - start;
+
+    if (elapsed < duration) {
+      // @ts-ignore
```

Comment:
Are the `@ts-ignore` still necessary?
I think it would be simpler to just make the `current` have a `number` type, and typecast it when returning.

Reply:
Unless there's something I've missed, it seems a lot simpler to use `number`s for everything instead of generics. Maybe when there's an update that allows `createTween` to tween between arbitrary types like strings or colours, there would be reason to revisit this. In any case, it fixes all the TS errors.

Reply:
I'm not sure why it was generic either, it's definitely easier to just remove it.
But I like the idea of making it tween arbitrary types.